ragna/assistants/_demo.py

```diff
@@ -39,16 +35,19 @@ def _markdown_answer(self) -> str:
             """
         ).strip()
 
-    def _default_answer(self, prompt: str, sources: list[Source]) -> str:
+    def _default_answer(self, messages: list[Message]) -> str:
+        prompt = messages[-1].content.strip()
         sources_display = []
-        for source in sources:
-            source_display = f"- {source.document.name}"
-            if source.location:
-                source_display += f", {source.location}"
-            source_display += f": {textwrap.shorten(source.content, width=100)}"
-            sources_display.append(source_display)
-        if len(sources) > 3:
-            sources_display.append("[...]")
+        for message in messages:
```

Comment:
Let's not overwhelm the user with information when using this demo assistant. Not sure if we even want to include any of the old messages. Maybe just add a small statistic like "before this question you asked N more".

Reply:
I like that a lot. I can make that change.